fsspec/implementations/http.py

```diff
@@ -110,25 +175,86 @@ def ls(self, url, detail=True, **kwargs):
         else:
             return list(sorted(out))
 
-    def cat(self, url):
-        r = self.session.get(url, **self.kwargs)
-        r.raise_for_status()
-        return r.content
+    ls = sync_wrapper(_ls)
```

Comment:
I got caught a couple of time when implementing a class with both sync and async method, blokgin myself by mistake. Personally I prefer to have a superclass that have no sync method and a superclass (or a utility function) that "syncify" them. That way I'm sure when working on  the superclass that I'm not going to call sync method in sync context.